### ISSUE TITLE
Scoreboard objective displayname cut off

### ISSUE THREAD

**Issue by lukasbableck** [ViaVersion/ViaBackwards#197]
Hello,
i have a small issue with scoreboards on my 1.15.2 server if I'm connecting with client version 1.12.2 or older. The displayname of the scoreboard objective is cut off.

Normally it should look like this: (1.15.2 client)
https://i.tmplrs.icu/RtWJ0v.png

But with anything lower than 1.13 it looks like this:
https://i.tmplrs.icu/m6TR-3.png

No errors in the console whatsoever.
I'm using the latest abstraction/dev builds of ViaVersion, ViaRewind and ViaBackwards on PaperSpigot 1.15.2.

/viaversion dump
https://dump.viaversion.com/2558cbf9e951c5421fcd58b19837474883b702df3b17ff1d243760cb6c9dffec

**Comment by KennyTV** [comment]
Nothing we can do about, 1.12 clients and lower allow a smaller amount of characters than 1.13+ do.
We already try to split them into prefix and suffix, so that the max length becomes 32 (realistically ~30 with split colors)

**Comment by lukasbableck** [comment]
I already had exactly this plugin working with exactly the same displayname while i was running a 1.8.8 server and using a 1.8.9 client. So if it works in 1.8, why doesn’t it work with clients older than 1.13 on a 1.15.2 server?

**Comment by KennyTV** [comment]
Pretty sure that's a no, unless they hacked something in the displayname rather than prefix and suffix. 1.8, too, disconnects with affixes longer than 16 characters

**Comment by lukasbableck** [comment]
The only thing I did in that plugin was setting the displayname of the scoreboard objective with Objective#setDisplayName.
Worked fine on the 1.8.8 server with 1.8 to 1.15.2 clients, works on the 1.15.2 server with clients newer than 1.12.2 but not for 1.12.2 and older.

**Comment by KennyTV** [comment]
Ah right, the displayname is also cut. You'd need to split the name yourself, if you want it fully displayed. Prefix and suffix are limited to 16, the displayname to 32 characters

**Comment by lukasbableck** [comment]
How do I do that then? I can only set the displayname with Objective#setDisplayName. As far as I know I cannot use prefixes there. Only for teams as far as I know.

**Comment by KennyTV** [comment]
Then you either need to use teams or live with the 32 character limit 👀

**Comment by lukasbableck** [comment]
Well, as I can't use teams in that place, I think i will have to find a solution on my own. It's definitely not a client limit as the client shows the displayname fine on an 1.8.8 server.

**Comment by KennyTV** [comment]
Well here you go, this is 1.10 MCP code from the team packet
```
            this.displayName = buf.readStringFromBuffer(32);
            this.prefix = buf.readStringFromBuffer(16);
            this.suffix = buf.readStringFromBuffer(16);
```
and this is the same for the 1.8 server. https://i.imgur.com/SPZ7KoE.png
(as it is for the Objective packet `this.b = var1.c(32);`)

**Comment by lukasbableck** [comment]
I found the issue. The issue was indeed ViaBackwards. My displayname is exactly 32 chars long, including color codes.
ViaBackwards has to convert the json format, which is used in the versions 1.13+, to text for the older versions. At this point a "§f" is added right at the beginning of the displayname. Because of that the String gets two chars longer and those two chars get cut at the end.

I fixed that behaviour in my version of ViaBackwards.
